xcell/mappers/mapper_Planck_base.py

```diff
@@ -48,10 +58,9 @@ def _get_mask(self):
                 msk = gp_mask
             else:
                 msk *= gp_mask
-        if self.file_ps_mask is not None:
+        if (self.file_ps_mask is not None) & (self.pre_mask_ps is False):
             for mode in self.ps_mask_mode:
-                field = self.ps_mask_modes[mode]
-                ps_mask = hp.read_map(self.file_ps_mask, field)
+                ps_mask = self._get_ps_mask()
```

Comment:
I think this shouldn't be inside the `for mode ...` loop

Reply:
you are right, the loop is already inside `_get_ps_mask()`